x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -466,11 +475,15 @@ export class TaskRunner<
       });
     }
 
-    let triggeredActions: AlertAction[] = [];
+    const alertExecutionStore: AlertExecutionStore = {
+      numberOfTriggeredActions: 0,
+      completion: ActionsCompletion.COMPLETE,
+    };
+
     if (!muteAll && this.shouldLogAndScheduleActionsForAlerts()) {
       const mutedAlertIdsSet = new Set(mutedInstanceIds);
 
-      const scheduledActionsForRecoveredAlerts = await scheduleActionsForRecoveredAlerts<
+      await scheduleActionsForRecoveredAlerts<
```

Comment:
Since we schedule actions for recovered alerts first, this brings up an interesting case where capping the max executed actions could potentially prevent active alerts from firing notifications if the limit is low and the number of recovered alerts with actions is high. Do we anticipate this being an issue? We're basically prioritizing recovered alerts above new/active alerts.

Reply:
Yes i also realised that and thought that it's ok... We have to stop execution if we reach the limit, don't we?
Would like to discuss with the team.

Reply:
>Good question in terms of priority. It does seem like the active alerting actions should take priority over recovered. We could defer and open a separate issue for this if we want to explore it further, but I'd be happy to change the ordering now

I'm +1 to what @pmuellr mentions, this is a good finding. It feels to me it is more important to notify of new alerts as fast as we can (and as much as we can) to shorten the MTTD as low as possible. I'd be ok if we change the order in general to queue new alert actions first so we get those notifications out ASAP.

Reply:
Done, change the order of active/recovered alert execution